### PR TITLE
Improve Discord Rich Presence

### PR DESCRIPTION
**Pull request by ficcialfaint**
### What does it change?
- Removes the second "In Main Menu" string
- Adds player counter instead of only max players (now it looks like **Players: 2/10**)
- You can now change the text shown in Rich Presence without changing the engine (Localization support)
- Makes the logo primary image when in lobby
- Makes Discord RPC automatically update when a player joins the server
### Media
image
image

### CODE REVIEW

File: Robust.Client/Utility/DiscordRichPresence.cs [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
         {
+            _defaultPresence = new()
+            {
+                State = _loc.GetString("discord-rpc-in-main-menu"),
+                Assets = new Assets
+                {
+                    LargeImageKey = "logo",
+                    LargeImageText = "I think coolsville SUCKS"
+                }
+            };
```
Comment (AJCM-git) [inline Robust.Client/Utility/DiscordRichPresence.cs:36]: didnt this used to also have Details and SmallImageKey

**Review by AJCM-git — CHANGES_REQUESTED** [review]
i dont see default files for the new ftl strings

File: Robust.Client/Utility/DiscordRichPresence.cs (hunk 1 above)
Comment (ficcialfaint) [reply Robust.Client/Utility/DiscordRichPresence.cs:36]: it is, but in my opinion there is no need in additional "in main menu" string
about the second image while in main menu: i just don't like that in main menu rich presence and on server look almost the same, so i changed it a bit 
pretty strange logic ik lol

**Comment by ficcialfaint** [comment]
> i dont see default files for the new ftl strings

you mean this space-wizards/space-station-14#14350 ?

**Comment by metalgearsloth** [comment]
> > i dont see default files for the new ftl strings
> 
> you mean this [space-wizards/space-station-14#14350](https://github.com/space-wizards/space-station-14/pull/14350) ?

This should be on engine.

**Comment by ficcialfaint** [comment]
> > > i dont see default files for the new ftl strings
> 
> > 
> 
> > you mean this [space-wizards/space-station-14#14350](https://github.com/space-wizards/space-station-14/pull/14350) ?
> 
> 
> 
> This should be on engine.

done

File: Resources/Locale/en-US/discordRPC.ftl [hunk 2]
```diff
@@ -0,0 +1,4 @@
+discord-rpc-in-main-menu = In Main Menu
+discord-rpc-character = Character: {$username}
```
Comment (mirrorcult) [inline Resources/Locale/en-US/discordRPC.ftl]: this isnt their character name, is it? pretty sure this is just their username, so it should be called something like that instead

**Review by mirrorcult — CHANGES_REQUESTED** [review]
is the PR linked in the desc still required for anything? its closed edit: nvm, i see the ftl file stuff

**Comment by ficcialfaint** [comment]
> is the PR linked in the desc still required for anything? its closed

Well, seems like it isn't since the default lines are already in the engine. I'll edit it

File: Resources/Locale/en-US/discordRPC.ftl (hunk 2 above)
Comment (ficcialfaint) [reply Resources/Locale/en-US/discordRPC.ftl]: Yea, it's a username, I dunno why it's called Character tho. Haven't touched it, leaved how it was. Do I change it?

Comment (mirrorcult) [reply Resources/Locale/en-US/discordRPC.ftl]: i'd say change it

Comment (ficcialfaint) [reply Resources/Locale/en-US/discordRPC.ftl]: done

**Review by mirrorcult — APPROVED** [review]

**Comment by ficcialfaint** [comment]
Thanks ^.^

**Pull request merged**